src/Processors/Transforms/WindowTransform.cpp

```diff
@@ -71,6 +74,9 @@ class IWindowFunction
         size_t function_index) const = 0;
 
     virtual std::optional<WindowFrame> getDefaultFrame() const { return {}; }
+
+    /// Is the frame type supported by this function.
+    virtual bool checkWindowFrameType(const WindowTransform * /*transform*/) const { return true; }
```

Comment:
Looks like a reference would be better here:
```
virtual bool checkWindowFrameType(const WindowTransform & /*transform*/) const { return true; }
```

Reply:
Though for consistency with `windowInsertResultInto()` it might be better to keep a pointer here.